pytorch_lightning/core/step_result.py

```diff
@@ -0,0 +1,336 @@
+from typing import Optional, Dict, Union, Sequence, Callable, MutableMapping, Any
+from torch import Tensor
+import torch
+from copy import copy
+
+
+class Result(Dict):
+
+    def __init__(
+            self,
+            minimize: Optional[Tensor] = None,
+            early_stop_on: Optional[Tensor] = None,
+            checkpoint_on: Union[Tensor, bool, None] = None,
+            hiddens: Optional[Tensor] = None,
+    ):
+
+        super().__init__()
+
+        if early_stop_on is not None:
+            self.early_stop_on = early_stop_on
+        if checkpoint_on is not None and checkpoint_on:
+            self.checkpoint_on = checkpoint_on
+        if hiddens is not None:
+            self.hiddens = hiddens
+        if minimize is not None:
+            err = 'Minimize can only be used in training_end, training_step_end, training_epoch_end'
```

Comment:
```suggestion
            err = 'Minimize can only be used in training_step, training_step_end, training_epoch_end'
```
Did you mean training_step?
And what is the meaning of minimize in training_epoch_end?